### ISSUE TITLE
Provide more options for response assertions

### ISSUE THREAD

**Issue by mrwatson** [kubeshop/tracetest#950]
**Is your feature request related to a problem? Please describe.**
Our potential use case would require in-depth assertions on JSON responses returned from tests. Assertions against trace data is helpful, but does not provide the full picture in our case. For example, if we had a structure returned that looked like this:

```json
{
  "id": "abc",
  "revision": 3,
  "title": "Demo Protocol Version 1",
  "fullVersion": {
    "major": 3,
    "minor": 2
  },
  "nested": {
    "id": "def",
    "revision": 4,
    "nestedAgain": {
      "id": "klm",
      "revision": 5,
      "owner": {
        "id": "demo"
      }
    },
    "identifier": "yyy"
  }
}
```

We would want to be able to write assertions of the form:
- `$.id` is a string
- `$.revision` is a number
- `$.nested.id` != `$.id`
- `$.nested.revision` is a number and does not match `$.revision`
- `$.nested.nestedAgain.owner.id` is a string matching 'demo'
- `$.nested.identifier` is a string of at least 3 characters

**Describe the solution you'd like**
Provide additional assertion types matching the use cases listed above.

**Describe alternatives you've considered**
Running a test from the command line and passing the data through JSONPath or some such tool to validate the response. This is certainly doable, but more difficult.

**Additional context**
N/A

**Comment by schoren** [comment]
I think we can add "filters" for attributes. For example

```
- tracetest.response.body | jsonpath $.identifier | length > 3 
- tracetest.response.body | jsonpath $.id | type = string
- tracetest.response.body | jsonpath $.nested.id != tracetest.response.body | jsonpath $.nested.id
```

This relates to https://github.com/kubeshop/tracetest/issues/782 too

**Comment by kdhamric** [comment]
@mathnogueira I believe we can close this. Can you review and close if you believe we have this capability based on the jsonpath work that was done?

**Comment by mathnogueira** [comment]
Yes, this was implemented back (probably v0.8). 

In that version, we build our assertion engine capable of retrieving data from a data source (environment variables, span attributes), applying fields to transform that data, and comparing it with another value.

The examples you used can be asserted as:

```python
# $.id is a string
attr:my_json | json_path '$.id' | type = "string"

# $.revision is a number
attr:my_json | json_path '$.revision' | type = "number"

# $.nested.id != $.id
attr:my_json | json_path '$.nested.id' != attr:my_json | json_path '$.id'

# $.nested.revision is a number and does not match $.revision
attr:my_json | json_path '$.nested.revision' | type = "number"
attr:my_json | json_path '$.nested.revision' != attr:my_json | json_path '$.revision'

# $.nested.nestedAgain.owner.id is a string matching 'demo'
attr:my_json | json_path '$.nested.nestedAgain.owner.id' | type = "string"
attr:my_json | json_path '$.nested.nestedAgain.owner.id' = "demo"

# $.nested.identifier is a string of at least 3 characters
attr:my_json | json_path '$.nested.identifier' | length >= 3
```

I'm closing this issue for now, if there is any question about it, please open a new one. And thanks a lot for opening this issue. It helped us a lot with the requirements for our assertion engine!
